File: Motocitizen/src/motocitizen/dictionary/AccidentStatus.java
package motocitizen.dictionary;

public enum AccidentStatus {
    /*
                            WHEN "acc_status_act" THEN "a"
                        WHEN "acc_status_dbl" THEN "d"
                        WHEN "acc_status_end"  THEN "e"
                        WHEN "acc_status_hide" THEN "h"
                        WHEN "acc_status_war" THEN "w"
     */
    ACTIVE("a", "Активный"),
    ENDED("e", "Отбой"),
    HIDDEN("h", "Скрыт"),
    CONFLICT("w", "Конфликт"),
    DUPLICATE("d", "Дубль");
//    ACTIVE("acc_status_act", "Активный"),
//    ENDED("acc_status_end", "Отбой"),
//    HIDDEN("acc_status_hide", "Скрыт"),
//    CONFLICT("acc_status_war", "Конфликт"),
//    DUPLICATE("acc_status_dbl", "Дубль");

    private final String code;
    private final String text;

    AccidentStatus(String code, String text) {
        this.code = code;
        this.text = text;
    }

    public static AccidentStatus parse(String status) {
        for (AccidentStatus a : AccidentStatus.values()) {
            if (a.code.equals(status)) return a;
        }
        return AccidentStatus.ACTIVE;
    }

    public String code() {
        return this.code;
    }

    public String string() {
        return this.text;
    }
}

File: Motocitizen/src/motocitizen/dictionary/AccidentStatus.kt
package motocitizen.dictionary

import com.fasterxml.jackson.annotation.JsonCreator

enum class AccidentStatus(override val code: String, override val text: String) : Dictionary<String> {
    ACTIVE("a", "Активный"),
    ENDED("e", "Отбой"),
    HIDDEN("h", "Скрыт");

    //    CONFLICT("w", "Конфликт"),
//    DUPLICATE("d", "Дубль");
    companion object {
        private val map = entries.associateBy { it.code }

        @Suppress("unused")
        @JsonCreator
        @JvmStatic
        fun fromString(code: String): AccidentStatus = map[code] ?: ACTIVE
    }
}